doc/Operation.md

```diff
@@ -0,0 +1,328 @@
+        // ... consumer-code
+    }
+}
+
+// Check is the Operation has created the value `T`
+if (!operationResult.HasValue())
+{
+    // The Operation completed but it couldn't build the value `T`. Most likely the operation stage ended with error or it was cancelled.
+    throw std::runtime_error("Operation finished with status:" + operationResult.Status());
+}
+
+// At this point, the Operation is success and we can get its value.
+auto valueT = operationResult.Value();
+```
+
+### PollUntilDone
+
+The `PollUntilDone()` is a convenience method for repeatedly call `Poll()` until the status is updated to a _final stage_. A `Response<T>` will be returned wrapping the model `T` and the Http raw response from which it was created. This method is useful when we are not interested about the Http raw response for each polling request.
+
+The `PollUntilDone()` method requires an `std::chrono::milliseconds` input parameter which controls how much time to wait between calling `Poll`.
+
+In the code below, instead of manually polling for updates, the `PollUntilDone` method is called.
+
+```cpp
+/**
+ * Consuming Operation<T> interface. Calling PollUntilDone.
+ *
+ */
+auto operationResult = client.ApiReturnOperation(...);
+
+try
+{
+    // Update each second
+    auto response = operationResult.PollUntilDone(std::chrono::milliseconds(1000));
+
+    // No need to check for `HasValue` here. The Operation will throw if the value can't be created after completing.
+    auto valueT = response.ExtractValue();
+    auto rawResponse = response.ExtractRawResponse();
+
+}
+catch (Azure::Core::RequestFailedException const& e)
+{
+    // Analyze the exception or re-throw.
+}
+```
+
+## Operation\<T> lifetime
+
+The `Operation<T>` holds an Http raw response. Each Operation\<T> derived class defines the moment when the Http raw response is moved to the Operation. This means that an Operation can be created without initially holding an Http raw response. In this case, calling `GetRawResponse` method from the Operation will `throw` as no http raw response has been set for it yet.
+
+The concrete Operation class defines since when the `GetRawResponse` can be called, we need to review each derived class to know at what time it is valid or not to call `GetRawResponse`.
+
+The code below show two examples of Operation derive classes. The first one requires an http raw response to be created, ensuring that `GetRawResponse` can be called at any moment after the Operation is created. The second class does not set an http raw response when it is init, which means that `GetRawResponse` can only be called after calling `Poll`, otherwise it will throw.
+
+```cpp
+/**
+ * Example of operations construction and `GetRawResponse` valid calls.
+ *
+ */
+
+// The operation can only be constructed based on an http raw response.
+class OperationA : public Azure::Core::Operation<Model> {
+public:
+    OperationA(Azure::Core::Http::RawResponse rawResponse);
```

Comment:
This doesn't seem right to me. Only the client lib should be able to construct an Operation<T> and it will most-likely NOT be created from a raw response. Customers don't need to know about this - move to guidelines.

Reply:
I was thinking that the `Operation<T>` gives a powerful feature and advanced feature to the users. They can basically create `Future requests` with it :) if they want.  But that's fine, I just wanted to make the general knowledge of this Class to have official documentation to support it.  I can move this else where